versions.lock

```diff
@@ -95,9 +96,9 @@ org.antlr:antlr4-runtime:4.7 (1 constraints: 7a0e125f)
 org.antlr:stringtemplate:3.2.1 (1 constraints: c10a3bc6)
 org.apache.ant:ant:1.9.1 (3 constraints: a721ed14)
 org.apache.ant:ant-launcher:1.9.1 (1 constraints: 69082485)
-org.apache.arrow:arrow-format:0.10.0 (1 constraints: 1f0de721)
-org.apache.arrow:arrow-memory:0.10.0 (1 constraints: 1f0de721)
-org.apache.arrow:arrow-vector:0.10.0 (1 constraints: e90c9734)
+org.apache.arrow:arrow-format:0.14.1 (1 constraints: 240df421)
+org.apache.arrow:arrow-memory:0.14.1 (1 constraints: 240df421)
+org.apache.arrow:arrow-vector:0.14.1 (2 constraints: 2012a545)
```

Comment:
0.15.1 is out. Does that version work?

Reply:
@samarthjain, did you see this question?

Reply:
Unfortunately in 0.15.1, they made vectors like DecimalVector, etc. final which prevents overriding the isSet() method to use NullabilityHolder. So going to stick to 0.14.1 for now.

Reply:
@samarthjain, as a follow-up, we should open a PR for Arrow that fixes the DecimalVector nullability problem so we don't need an implementation here in Iceberg.

Reply:
Yeah, it would be great to see an Arrow patch for what would be the appropriate fix.